From 2b3dd57a6ccb2940f2e9119ae04e14362e2a1f61 Mon Sep 17 00:00:00 2001
From: Remi Collet <fedora@famillecollet.com>
Date: Fri, 22 Jul 2016 08:14:12 +0200
Subject: [PATCH] Fix gd2/gd2_read.c:8:6: error: 'error' may be used
 uninitialized in this function [-Werror=maybe-uninitialized]

Also report about any error, not only the last one.
---
 tests/gd2/gd2_read.c | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/tests/gd2/gd2_read.c b/tests/gd2/gd2_read.c
index 94fe069..8ce8bd1 100644
--- a/tests/gd2/gd2_read.c
+++ b/tests/gd2/gd2_read.c
@@ -5,7 +5,7 @@
 
 int main(int argc, char *argv[])
 {
-	int error, i = 0;
+	int error = 0, i = 0;
 	gdImagePtr im, exp;
 	FILE *fp;
 	char *path[] = {
@@ -40,8 +40,6 @@ int main(int argc, char *argv[])
 				gdTestErrorMsg("image %s differs from expected result\n", path[i]);
 				gdImageDestroy(im);
 				error = 1;
-			} else {
-				error = 0;
 			}
 			if (exp) {
 				gdImageDestroy(exp);
@@ -52,8 +50,6 @@ int main(int argc, char *argv[])
 				gdTestErrorMsg("image %s should have failed to be loaded\n", path[i]);
 				gdImageDestroy(im);
 				error = 1;
-			} else {
-				error = 0;
 			}
 		}
 		i++;